lib/services/presets.go

```diff
@@ -561,6 +562,111 @@ func NewSystemOktaRequesterRole() types.Role {
 	return role
 }
 
+// NewSystemIdentityCenterAccessRole creates a role that allows access to AWS
+// IdentityCenter resources via Access Requests
+func NewSystemIdentityCenterAccessRole() types.Role {
+	if modules.GetModules().BuildType() != modules.BuildEnterprise {
+		return nil
+	}
+
+	return &types.RoleV6{
+		Kind:    types.KindRole,
+		Version: types.V7,
+		Metadata: types.Metadata{
+			Name:        teleport.SystemIdentityCenterAccessRoleName,
+			Namespace:   apidefaults.Namespace,
+			Description: "Access AWS IdentityCenter resources",
+			Labels: map[string]string{
+				types.TeleportInternalResourceType: types.SystemResource,
+				types.OriginLabel:                  common.OriginAWSIdentityCenter,
+			},
+		},
+		Spec: types.RoleSpecV6{
+			Allow: types.RoleConditions{
+				AccountAssignmentLabels: types.Labels{
+					types.OriginLabel: []string{common.OriginAWSIdentityCenter},
+				},
+				Rules: []types.Rule{
+					types.NewRule(types.KindIdentityCenter, RO()),
+				},
+			},
+		},
+	}
+}
+
+// NewSystemIdentityCenterRequesterRole creates a role that allows a user to
+// request access to AWS IdentityCenter resources via Access Requests
+func NewSystemIdentityCenterRequesterRole() types.Role {
+	if modules.GetModules().BuildType() != modules.BuildEnterprise {
+		return nil
+	}
+
+	return &types.RoleV6{
+		Kind:    types.KindRole,
+		Version: types.V7,
+		Metadata: types.Metadata{
+			Name:        teleport.SystemIdentityCenterRequesterRoleName,
+			Namespace:   apidefaults.Namespace,
+			Description: "Request AWS IdentityCenter resources",
+			Labels: map[string]string{
+				types.TeleportInternalResourceType: types.SystemResource,
+				types.OriginLabel:                  common.OriginAWSIdentityCenter,
+			},
+		},
+		Spec: types.RoleSpecV6{
+			Allow: types.RoleConditions{
+				Request: &types.AccessRequestConditions{
+					Roles: []string{
+						teleport.SystemIdentityCenterAccessRoleName,
+					},
```

Comment:
wonder why we need the separate AWS IC requester role for that 

After taking a look at our code we can just extend the NewPresetRequesterRole that currently grands access request role for PresetGroupAccessRoleName and Okta:  
https://github.com/gravitational/teleport/blob/94dc1a90274aff69e104cb0b8fad231434792001/lib/services/presets.go#L706

Does reusing `NewPresetReviewerRole` and NewPresetRequesterRole  this will work in AWS IC  case ?